apps/test/unit/templates/courseRollupPages/RollupLessonEntrySectionTest.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import {expect} from '../../../util/reconfiguredChai';
+import RollupLessonEntrySection from '@cdo/apps/templates/courseRollupPages/RollupLessonEntrySection';
+import {courseData} from './rollupTestData';
+import i18n from '@cdo/locale';
+import {studio} from '@cdo/apps/lib/util/urlHelpers';
+import {Provider} from 'react-redux/src';
+import {getStore} from '@cdo/apps/redux';
+import EnhancedSafeMarkdown from '@cdo/apps/templates/EnhancedSafeMarkdown';
+import InlineMarkdown from '@cdo/apps/templates/InlineMarkdown';
+
+describe('RollupLessonEntrySection', () => {
+  let defaultProps;
+  beforeEach(() => {
+    defaultProps = {
+      objectToRollUp: 'Resources',
+      lesson: courseData.units[0].lessons[0]
+    };
+  });
+
+  it('renders list of resources when there are resources', () => {
+    const wrapper = mount(<RollupLessonEntrySection {...defaultProps} />);
+
+    expect(wrapper.find('ResourceList').length).to.equal(2);
+  });
+
+  it('renders no resources message when no resources', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        lesson={courseData.units[1].lessons[0]}
+      />
+    );
+
+    expect(wrapper.containsMatchingElement(<p>{i18n.rollupNoResources()}</p>))
+      .to.be.true;
+  });
+
+  it('renders list of prep when there is prep', () => {
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <RollupLessonEntrySection {...defaultProps} objectToRollUp={'Prep'} />
+      </Provider>
+    );
+
+    expect(
+      wrapper.containsMatchingElement(
+        <EnhancedSafeMarkdown
+          markdown={defaultProps.lesson.preparation}
+          expandableImages
+        />
+      )
+    ).to.be.true;
+  });
+
+  it('renders no prep message when no prep', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        objectToRollUp={'Prep'}
+        lesson={courseData.units[1].lessons[0]}
+      />
+    );
+
+    expect(wrapper.containsMatchingElement(<p>{i18n.rollupNoPrep()}</p>)).to.be
+      .true;
+  });
+
+  it('renders list of vocab when there is vocab', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        objectToRollUp={'Vocabulary'}
+      />
+    );
+
+    expect(
+      wrapper.containsMatchingElement(
+        <li key={defaultProps.lesson.vocabularies[0].key}>
+          <InlineMarkdown
+            markdown={`**${defaultProps.lesson.vocabularies[0].word}** - ${
+              defaultProps.lesson.vocabularies[0].definition
```

Comment:
this seems too specific. I wouldn't consider the component to be broken if the formatting of this vocab changed, or if we switched to EnhancedSafeMarkdown, or if we stopped using `li` elements for the list. I would suggest instead checking that both the vocab word and the definition both appear in the output. one way would be to check that they appear in wrapper.text(), but other ways could work too. if we want to make sure there are no regressions in how the page looks, we can use an eyes test for that.